go/vt/sqlparser/analyzer.go

```diff
@@ -72,7 +72,13 @@ func Preview(sql string) int {
 	case "delete":
 		return StmtDelete
 	}
-	switch strings.ToLower(trimmed) {
+	// For the following statements is not enough to just check
+	// loweredFirstWord. This is because they are not statements
+	// in the grammar and we are relying in Preview to parse them.
```

Comment:
minor nits:
"relying on" not "relying in".
I might clarify your counterexample as "BEGIN JUNK" as opposed to "BEGIN ..."

Otherwise LGTM